pkgs/development/python-modules/segyio/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, python
+, scikit-build
+, pytest
+, numpy
+}:
+
+stdenv.mkDerivation rec {
+  pname = "segyio";
+  version = "1.9.9";
+
+  postPatch = ''
+    # Removing unecessary build dependency
+    substituteInPlace python/setup.py --replace "'pytest-runner'," ""
+
+    # Fixing bug: This MINSIGSTKSZ is nowhere to be found inside the code base, so I removed this expression
+    substituteInPlace external/catch2/catch/catch.hpp --replace "32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ" 32768
+  '';
+
+  src = fetchFromGitHub {
+    owner = "equinor";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-L3u5BHS5tARS2aIiQbumADkuzw1Aw4Yuav8H8tRNYNg=";
+  };
+
+  nativeBuildInputs = [ cmake ninja python ];
+  buildInputs = [ scikit-build ];
+
+  doCheck = true;
+  checkInputs = [ pytest numpy ];
```

Comment:
```suggestion
  checkInputs = [ pytestCheckHook numpy ];
```

Reply:
Thx for the review! Lemme test it.

Reply:
It throws
```
============================= test session starts ==============================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /build/source/build
collected 0 items                                                              

============================ no tests ran in 0.05s =============================
error: builder for '/nix/store/a06jfbkfqahb677n4kqwsra0dnfpm0j1-segyio-1.9.9.drv' failed with exit code 5;
       last 10 log lines:
       > patching script interpreter paths in /nix/store/ikl65csfck1h4blcvwd3bc9yzpavkfxk-segyio-1.9.9
       > checking for references to /build/ in /nix/store/ikl65csfck1h4blcvwd3bc9yzpavkfxk-segyio-1.9.9...
       > pytestCheckPhase
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0
       > rootdir: /build/source/build
       > collected 0 items                                                              
       >
       > ============================ no tests ran in 0.05s =============================
       For full logs, run 'nix log /nix/store/a06jfbkfqahb677n4kqwsra0dnfpm0j1-segyio-1.9.9.drv'.
 ```

Reply:
I assume that the pytestCheckHook tries test as normally is done like `pytest tests/` in the root of the repository, but it doesn't find any tests to be done. Nevertheless, that is bound not to work since the pytest is called withing the cmake build. I just put pytest as a checkInput for it to have it available during the tests execution. In fact I have tried to overwrite the checkPhase like this:
```
checkPhase = "pytest tests/"
```
and the same thing happens, it cannot find any tests. When I leave the checkPhase untouched the tests are correctly ran, and everything works.

Reply:
> Nevertheless, that is bound not to work since the pytest is called withing the cmake build. 

Ok, this works correctly them.

Reply:
Please add a comment with that information.

Reply:
Aye sir!